libavfilter/dnn/dnn_backend_common.c

```diff
@@ -69,3 +69,66 @@ DNNReturnType ff_dnn_fill_task(TaskItem *task, DNNExecBaseParams *exec_params, v
 
     return DNN_SUCCESS;
 }
+
+static void *async_thread_routine(void *args)
+{
+    DNNAsyncExecModule *async_module = args;
+    void *request = async_module->args;
+
+    async_module->start_inference(request);
+    async_module->callback(request);
+#if HAVE_PTHREAD_CANCEL
+    pthread_exit(0);
```

Comment:
i'm not sure if pthread_exit is needed here, just google it and do not find the result.

Reply:
Not necessary. According to [this](https://man7.org/linux/man-pages/man3/pthread_create.3.html), this is equivalent to `return 0`.


> The new thread terminates in one of the following ways:
> 
> 1. It calls pthread_exit(3), specifying an exit status value that is > available to another thread in the same process that calls > pthread_join(3).
> 2. It returns from start_routine().  This is equivalent to calling pthread_exit(3) with the value supplied in the return statement.

I didn't know FFmpeg works on windows as well, so I included it. I'll remove it. Thanks